### ISSUE TITLE
OSC-related examples don't compile

### ISSUE THREAD

**Issue by cerkut** [openframeworks/openFrameworks#3574]
The following examples, which all depend on ofxOSC do not compile on Yosemite OS X 10.10 (Xcode                                   6.1.1, Build 6A2008a:

- addons/oscChatSystemExample/
- addons/oscReceiveExample
- addons/oscSenderExample
- gui/oscParametersSender/
- gui/oscParametersReceiver

For example, the last one gives

Undefined symbols for architecture i386:
  "GetHostByName(char const*)", referenced from:
      IpEndpointName::GetHostByName(char const*) in IpEndpointName.o
  "SocketReceiveMultiplexer::SocketReceiveMultiplexer()", referenced from:
      UdpListeningReceiveSocket::UdpListeningReceiveSocket(IpEndpointName const&, PacketListener*, bool) in ofxOscReceiver.o
  "SocketReceiveMultiplexer::DetachSocketListener(UdpSocket*, PacketListener*)", referenced from:
      UdpListeningReceiveSocket::~UdpListeningReceiveSocket() in ofxOscReceiver.o
  "SocketReceiveMultiplexer::AttachSocketListener(UdpSocket*, PacketListener*)", referenced from:
      UdpListeningReceiveSocket::UdpListeningReceiveSocket(IpEndpointName const&, PacketListener*, bool) in ofxOscReceiver.o
  "SocketReceiveMultiplexer::Run()", referenced from:
      UdpListeningReceiveSocket::Run() in ofxOscReceiver.o
  "UdpSocket::Bind(IpEndpointName const&, bool)", referenced from:
      UdpListeningReceiveSocket::UdpListeningReceiveSocket(IpEndpointName const&, PacketListener*, bool) in ofxOscReceiver.o
  "UdpSocket::Connect(IpEndpointName const&, bool)", referenced from:
      UdpTransmitSocket::UdpTransmitSocket(IpEndpointName const&, bool) in ofxOscSender.o
  "typeinfo for UdpSocket", referenced from:
      typeinfo for UdpTransmitSocket in ofxOscSender.o
      typeinfo for UdpListeningReceiveSocket in ofxOscReceiver.o
  "SocketReceiveMultiplexer::AsynchronousBreak()", referenced from:
      UdpListeningReceiveSocket::AsynchronousBreak() in ofxOscReceiver.o
  "UdpSocket::UdpSocket()", referenced from:
      UdpTransmitSocket::UdpTransmitSocket(IpEndpointName const&, bool) in ofxOscSender.o
      UdpListeningReceiveSocket::UdpListeningReceiveSocket(IpEndpointName const&, PacketListener*, bool) in ofxOscReceiver.o
  "UdpSocket::SetUdpBufferSize(unsigned long)", referenced from:
      ofxOscSender::setup(std::string, int, bool) in ofxOscSender.o
      ofxOscReceiver::setup(int, bool) in ofxOscReceiver.o
  "UdpSocket::Send(char const*, unsigned long)", referenced from:
      ofxOscSender::sendBundle(ofxOscBundle&) in ofxOscSender.o
      ofxOscSender::sendMessage(ofxOscMessage&, bool) in ofxOscSender.o
  "UdpSocket::GetUdpBufferSize()", referenced from:
      ofxOscSender::setup(std::string, int, bool) in ofxOscSender.o
      ofxOscReceiver::setup(int, bool) in ofxOscReceiver.o
  "SocketReceiveMultiplexer::~SocketReceiveMultiplexer()", referenced from:
      UdpListeningReceiveSocket::UdpListeningReceiveSocket(IpEndpointName const&, PacketListener*, bool) in ofxOscReceiver.o
      UdpListeningReceiveSocket::~UdpListeningReceiveSocket() in ofxOscReceiver.o
  "UdpSocket::~UdpSocket()", referenced from:
      UdpTransmitSocket::UdpTransmitSocket(IpEndpointName const&, bool) in ofxOscSender.o
      UdpTransmitSocket::~UdpTransmitSocket() in ofxOscSender.o
      UdpListeningReceiveSocket::UdpListeningReceiveSocket(IpEndpointName const&, PacketListener*, bool) in ofxOscReceiver.o
      UdpListeningReceiveSocket::~UdpListeningReceiveSocket() in ofxOscReceiver.o
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by admsyn** [comment]
@cerkut This would happen if your project is including the oscpack src folder, but ignoring the "posix" folder which includes the missing function implementations. If you manually add the posix folder it goes fine (if the header paths are also set up right). 

Did you happen to add ofxOsc with OFPlugin? This looks like a bug in its platform-specific folder selection.

**Comment by cerkut** [comment]
Nice, double spot on! I have compiled some OSC examples from the of_v0.8.4_osx_release, and clearly see the win32 and posix directories under ip.

I have added ofxOsc with OFPlugin, which created oscpack src and ip directories, but not win32 and posix directories under ip. It also added only src to header search paths. I'll create an issue there, closing this one. Thanks for looking into this.

**Comment by cerkut** [comment]
Now in the examples below, the oscpack/src/ip contains both the posix and win32, as folders and in header search paths

- addons/oscChatSystemExample/
- addons/oscReceiveExample
- addons/oscSenderExample

These examples don't compile and break /scripts/osx/buildAllExamples.sh

Removing the win32 dirs on Xcode project is the easiest fix.

**Comment by bilderbuchi** [comment]
on which OF version is this? how did you get/create the xcode project files of these examples?

**Comment by cerkut** [comment]
Good points. I usually copy the project files from the latest release to the examples of the latest git.

In this case, however, I have worked with previous project files which are not overwritten by git fetch & pull.

I have now compiled the 0.8.4 examples by using scripts/osx/buildAllExamples.sh now without a problem.

It is likely that the problem was in the project files, I’ll close the issue.

Thanks
